### PR TITLE
Update embedded CFamily analyzer to v6.26.0.36731

### PR DESCRIPTION
**Pull request by Godin**
Release notes:
* `6.22` - https://jira.sonarsource.com/secure/ReleaseNote.jspa?projectId=10090&version=16806
* `6.23` - https://jira.sonarsource.com/secure/ReleaseNote.jspa?projectId=10090&version=16855
* `6.24` - https://jira.sonarsource.com/secure/ReleaseNote.jspa?projectId=10090&version=16866
* `6.25` - https://jira.sonarsource.com/secure/ReleaseNote.jspa?projectId=10090&version=16938
* `6.26` - https://jira.sonarsource.com/secure/ReleaseNote.jspa?projectId=10090&version=16932

### CODE REVIEW

**Comment by duncanp-sonar** [comment]
Thanks for this @Godin.

[CPP-3119](https://jira.sonarsource.com/browse/CPP-3119) in v6.22 required some changes to the protocol, which @rita-sonarsource made in our working branch `branch-cmake`. So, we'll merge this in to `branch-cmake` rather than `master`.

Presumably there are no other protocol changes we need to take account of? Thanks.

**Comment by Godin** [comment]
> [CPP-3119](https://jira.sonarsource.com/browse/CPP-3119) in v6.22 required some changes to the protocol

@duncanp-sonar I wouldn't name this "changes to the protocol" since this does not change how data is sent :wink:
However indeed this new boolean flag should have value that matches Visual Studio Project configuration,
and AFAIK if not set, then we'll simply have behaviour of previous version(s).

Changes made by @rita-sonarsource in https://github.com/SonarSource/sonarlint-visualstudio/pull/2524 regarding this [CPP-3119](https://jira.sonarsource.com/browse/CPP-3119) and [CPP-3079](https://jira.sonarsource.com/browse/CPP-3079) look good to me 👍 but maybe will be wise to also double-check with @abbas-sabra-sonarsource who definitely has more expertise about these flags than me 😉 

> Presumably there are no other protocol changes we need to take account of? 

There is definitely no changes in how data is sent.

And regarding new flags/options AFAIK there were only the two above ones - I double-checked

```
git log 6.21.0.32709..6.25.0.35661 -- sonar-cfamily-plugin/src/main/java/com/sonar/cpp/analyzer/MsvcDriver.java
```

There was [CPP-3206](https://jira.sonarsource.com/browse/CPP-3206) that changed way of handling `INCLUDE` environment variable -   that is https://github.com/SonarSource/sonarlint-visualstudio/blob/v4.37.0.34839/src/Integration.Vsix/CFamily/PortedFromJava/MsvcDriver.cs#L294
 in your codebase. But this doesn't seem to be relevant to Visual Studio, because of https://github.com/SonarSource/sonarlint-visualstudio/blob/v4.37.0.34839/src/Integration.Vsix/CFamily/CFamilyHelper_Capture.cs#L70 
Similarly maybe better to double-check all this with @abbas-sabra-sonarsource

**Comment by abbas-sabra-sonarsource** [comment]
@Godin @duncanp-sonar, I confirm.
I previously discussed and reviewed @rita-sonarsource changes from 6.21 to 6.23 and there is nothing to update on SLVS side from 6.23 to 6.25.

File: src/EmbeddedSonarAnalyzer.props [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <!-- Note: don't forget to upload the new packages to the package feed when changing 
          the SonarAnalyzer version -->
     <EmbeddedSonarAnalyzerVersion>8.26.0.34506</EmbeddedSonarAnalyzerVersion>
-    <EmbeddedSonarCFamilyAnalyzerVersion>6.23.0.34138</EmbeddedSonarCFamilyAnalyzerVersion>
+    <EmbeddedSonarCFamilyAnalyzerVersion>6.25.0.35661</EmbeddedSonarCFamilyAnalyzerVersion>
```
Comment (abbas-sabra-sonarsource) [inline src/EmbeddedSonarAnalyzer.props:9]:
```suggestion
    <EmbeddedSonarCFamilyAnalyzerVersion>6.26.0.36731</EmbeddedSonarCFamilyAnalyzerVersion>
```

File: src/Integration.Vsix.UnitTests/CFamily/CFamilyEmbeddedSonarWayRulesTests.cs [hunk 2]
```diff
@@ -40,12 +40,12 @@ public class CFamilyEmbeddedSonarWayRulesTests
...
         // 5. Repeat for C++.
 
-        // Rule data for C-Family plugin v6.23 (build 34138)
+        // Rule data for C-Family plugin v6.25 (build 35661)
         private const int Active_C_Rules = 205;
         private const int Inactive_C_Rules = 103;
 
-        private const int Active_CPP_Rules = 388;
-        private const int Inactive_CPP_Rules = 161;
+        private const int Active_CPP_Rules = 391;
```
Comment (abbas-sabra-sonarsource) [inline src/Integration.Vsix.UnitTests/CFamily/CFamilyEmbeddedSonarWayRulesTests.cs:47]:
```suggestion
        private const int Active_CPP_Rules = 395;
```

**Review by abbas-sabra-sonarsource — COMMENTED** [review]
update to 6.26 directly

**Comment by abbas-sabra-sonarsource** [comment]
@duncanp-sonar This should be ready to be consumed. It includes the fix for reproducer/preamble.

**Review by duncanp-sonar — APPROVED** [review]

**Pull request merged**